lib/utils.js

```diff
@@ -0,0 +1,21 @@
+const parser = require('@asyncapi/parser');
+
+function validate(asyncApiDocs) {
+  return new Promise((resolve,reject) => {
+    const parsedDocs = [];
+    const Promises = [];
+    asyncApiDocs.forEach(document => {
+      Promises.push(new Promise((resolve, reject) => {
+        parser.parse(document.json())
+          .then(parsedDoc => parsedDocs.push(parsedDoc))
+          .then(() => resolve())
+          .catch(err => reject(err));
+      }));
+    });
+    Promise.all(Promises)
+      .then(() => resolve(parsedDocs))
+      .catch(err => reject(err));
+  });
```

Comment:
You can use simple `map` function. 

```suggestion
return new Promise.all(asyncApiDocs.map(async doc => parser.parse(document.json())));
```

`Promise.all` returns the single `Promise` (and when is fullfiled array of data) and also handles the rejection for each promise inside input array, so you don't need to handle them in you side :) As you can see code is much simpler to read.

Reply:
Totally agree, never thought like that :)